Test/NumberTheory/FastPower.test.cpp

```diff
@@ -0,0 +1,28 @@
+#define FAST_POWER_TEST
+#define CATCH_CONFIG_MAIN
+
+#include "../catch.hpp"
+#include "../../NumberTheory/FastPower.cpp"
+
+
+TEST_CASE("Base cases", "[fast_power]") {
+    //REQUIRE(fast_power(0,0) == undefined);
+    REQUIRE(fast_power(0,1) == 0);
+    REQUIRE(fast_power(1,0) == 1);
+    REQUIRE(fast_power(1,1) == 1);
+}
+
+TEST_CASE("Normal cases", "[fast_power]") {
+    REQUIRE(fast_power(2,2) == 4);
+    REQUIRE(fast_power(2,4) == 16);
+    REQUIRE(fast_power(3,4) == 81);
+    REQUIRE(fast_power(7,9) == 40353607);
+    REQUIRE(fast_power(15,10) == 576650390625);
+}
+
+TEST_CASE("Overflow cases", "[fast_power]") {
+    REQUIRE(fast_power(2,100) == 68719476736);
+    REQUIRE(fast_power(10,99) == 4440381706574496940U);
```

Comment:
I thought about this for a while and I think what we need to do is check how many digits the result is going to take. That would be done as follows:
```
digits_required = floor(exponent * log10(base)) + 1
```
If `digits_required > 19`, then use modulo `1000000000007` and tell the user about this too. Otherwise, we can directly use the result.